test/test_tags.cpp

```diff
@@ -150,76 +148,123 @@ struct Matches testmatches[] = {
       { "not(empty:/^$/)", mtev_false },
       { "not(empty:)", mtev_false },
       { "not(empty)", mtev_false },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "b\"KGZvbyk=\":bar",
-    mtev_true,
     {
       { "and(*:*)", mtev_true },
       { "and(*:bar)", mtev_true },
       { "and(foo:bar)", mtev_false },
       { "and(*:/ba(.?)r/)", mtev_true },
       { "and(b\"KGZvbyk=\":bar)", mtev_true },
       { "and(b/XChm/:bar)", mtev_true },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "b\"W2Jhcl0=\":b\"Kipmb28qKg==\"",
-    mtev_true,
     {
       { "and(b\"W2Jhcl0=\":*)", mtev_true },
       { "and(*:b!Kipmb28qKg==!)", mtev_true },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "b\"Kipmb28qKg==\":b\"W2Jhcl0=\"",
-    mtev_true,
     {
       { "and(*:b\"W2Jhcl0=\")", mtev_true },
       { "and(b\"Kipmb28qKg==\":*)", mtev_true },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "b\"P2Zvbz8=\":b\"W2Jhcl0=\"",
-    mtev_true,
     {
       { "and(*:b\"W2Jhcl0=\")", mtev_true },
       { "and(b\"P2Yq\":*)", mtev_true },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "b\"Lz9oaXN0b2dyYW0vKC4rKSQp\":quux",
-    mtev_true,
     {
       { "and(*:*)", mtev_true },
       { "and(hint(*:*))", mtev_true },
       { "and(hint(*:*,index:none))", mtev_true },
       { "hint(and(b!Lz9oaXN0b2dyYW0vKC4rKSQp!:quux),foo:bar)", mtev_true },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "b\"L2Zvby8oXig/OlswLTldezJ9LSkvKC4rKSQp\":bar",
-    mtev_true,
     {
       { "and(*:bar)", mtev_true },
       { "and(b\"L2Zvby8oXig/OlswLTldezJ9LSkvKC4rKSQp\":bar)", mtev_false },
       { "and(b!L2Zvby8oXig/OlswLTldezJ9LSkvKC4rKSQp!:bar)", mtev_true },
-      { NULL, 0 }
+      { NULL, mtev_false }
     }
   },
   {
     "tag:this_tag_pair_is_too_long_0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
     "0000000000000000000000000000000257",
-    mtev_false,
     {
-      { NULL, 0 }
+      { "and(tag:*too_long*)", mtev_false },
+      { NULL, mtev_false }
+    }
+  },
+  {
+    "tag1:value1,"
+    "tag2:value2,"
+    "tag3:this_is_the_max_length_of_a_tag_pair_000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```

Comment:
I don't understand why these strings are not generated in code.  If the constant's value changes, these will fail.  The structure this belongs to can just be created on the heap and the string itself filled in code.

This was already mentioned before in comments

Reply:
whoops. I missed this one. I'll fix it 👍